google/ggrc-core
# GGRC-3962/GGRC-4051/GGRC-4097/GGRC-4177/GGRC-4185/GGRC-4178 FE Revision History - Phase1

# Dependencies

This PR is `on hold` until the dependencies are merged:

- [x] #6891

# Issue description

In a lot of cases, Admin(s) of objects (for example, Control) doesn't want anybody to modify his object.

To solve this problem 'Revisions (Versions)' feature will be introduced. In the result, any user with READ rights to an object will be able to propose changes. Only user with EDIT rights will be able to approve / decline such proposed revisions. Moreover, admin(s) of objects will be informed via notifications, that revision(s) have been proposed.

The access in application is going to be restricted hardly, when 'Revisions (Versions)' feature will be delivered. Most of users will receive global role = Creator. In most cases only Admins will be able to updated their objects and approve / decline / restore revisions (versions.)

Revisions feature should be added to all snapshottable objects, but Controls and Risks are with higher priority.

# Steps to test the changes

1) Open Control or Risk
2) You can see new tabs: '_Change Proposals_' and '_Revision History_'

# Sanity checklist




# PR Review checklist
